DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/src/golden_pca.hpp

```diff
@@ -0,0 +1,316 @@
+#include <math.h>
+
+#include <fstream>
+#include <iomanip>
+#include <random>
+
+/*
+This file implements the steps to
+identify principal components (Eigen vectors)
+of a matrix and finally transform input matrix
+along the directions of the principal components
+
+Following are the main steps in order to transform an
+matrix A made of multiple samples with different features.
+The matrix A will contain n samples with p features
+making it an n row, p columns matrix
+
+Here are the steps performed by this file:
+1. Compute the covariance matrices of these matrices
+2. Compute Eigen vectors and Eigen values of the covariance matrices using the
+QR iteration method
+
+*/
+template <typename T>
+class GoldenPCA {
```

Comment:
PCA? I guess you meant SVD? Or maybe you didn't mean to add this file?

Reply:
I copy the golden PCA to obtain the eigen values for reference, so it technically is PCA